<launch>
  <arg name="debug" default="false"/>
  <arg if="$(arg debug)" name="launch_prefix" default="gdb -ex run --args"/>
  <arg unless="$(arg debug)" name="launch_prefix" default=""/>

  <arg name="prefix" default="robot_"/>	
  <arg name="rs_lidar_config_path" default="$(find rslidar_sdk)/config/config.yaml"/>
  <arg name="frame_id" default="$(arg prefix)3d_laser_link" />
  <arg name="msop_port" default="6699" />
  <arg name="difop_port" default="7788" />
  <arg name="lidar_type" default="RSBP"/>
  <arg name="node_name" default="lidar_3d"/>
  <arg name="scan_topic" default="points_raw"/>
  <arg name="nodelet_name" default="rslidar_nodelet"/>

  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_name)" args="manager" output="screen"  launch-prefix="$(arg launch_prefix)">
  </node>
  <node pkg="nodelet" type="nodelet" name="rslidarNodelet" args="load rslidar_sdk/rslidarNodelet $(arg nodelet_name)" output="screen">
    <param name="lidar_type" value="$(arg lidar_type)"/>
    <param name="msop_port" value="$(arg msop_port)"/>
    <param name="difop_port" value="$(arg difop_port)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="rs_lidar_config_path" value="$(arg rs_lidar_config_path)"/>
    <param name="node_name" value="$(arg node_name)"/>
    <param name="scan_topic" value="$(arg scan_topic)"/>
    <remap from="/rslidar_points" to="$(arg node_name)/$(arg scan_topic)"/>
  </node>
</launch>
